### PR TITLE
Karazhan rework

### LINKED ISSUE #173: Karazhan The opera house Script Bug
**Description**:
The scene can't trigger
http://www.wowhead.com/npc=16812/barnes
**Current behaviour**: Tell us what happens

**Expected behaviour**: Tell us what should happen instead

**Steps to reproduce the problem**:
1. Click on the NPC after didn't open the option.

**Branch(es)**: 0.x / 1.x / master (Specify the branch(es) affected by this issue) 

**AC hash/commit**:  

**Operating system**:  
WIN7/X64
**Modules**:  Are you using modules? if yes, list them

**Other Customizations**: Are you use core/db patches or other customizations? If yes please specify them here.

##

### PR DESCRIPTION
**Pull request by talamortis**
**Changes proposed:**
**Opera Event**
- [ ] Fix loot tables for Opera event (RAJ)
- [x] Opera events working correctly
- [x] The crone cyclone will now move instead of stood in 1 place.
- [x] The Crone chain lightning will now cast more often

**Nightbane**
- [ ] Change Intro path of Nightbane
- [x] Nightbane will now correctly work

**Shade of Aran**
- [x] Shade of Aran will now summon water elementals in the correct place.

**Moroes**
- [x] Moroes will attack as he comes out of Vanish

**Illhoof**
- [x] Illhoof will now spawn back his Imp after 45 seconds

**Prince Malcazhaar**
- [x] Prince will now spawn only 1 axe instead of two.
- [x] Prince axe will now attack with both weapons instead of just one.
- [x] Prince Axe will now change target onto random, instead of focusing one player (fix on next push)
- [ ] Fix Enfeeble for Prince.
- [x] Rescript Prince Malcazhaar 
- [x] Make Infernal Relay Cast infernal

**Target branch(es):** master

**Issues addressed:** Closes #173 #172 #174 #39 

**Tests performed:** (Does it build, tested in-game, etc)
Tested in game all working correctly.

**NOTE** You no longer need to squash your commits, on merge we will squash it for you. (GitHub added a new feature)

### CODE REVIEW

File: src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /*
- * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: http://github.com/azerothcore/azerothcore-wotlk/LICENSE-GPL2
- * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
- * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
- */
+* Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: http://github.com/azerothcore/azerothcore-wotlk/LICENSE-GPL2
+* Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+* Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
+*/
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp]: ?

File: src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp [hunk 2]
```diff
@@ -422,19 +426,18 @@ class boss_malchezaar : public CreatureScript
                     Talk(SAY_AXE_TOSS2);
 
                     Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100, true);
-                    for (uint8 i = 0; i < 2; ++i)
                     {
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp]: what does this change mean? why removing the ``for`` without removing also its block?

File: src/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp [hunk 3]
```diff
@@ -296,94 +300,158 @@ class boss_shade_of_aran : public CreatureScript
                     }
                 }
                 NormalCastTimer = 1000;
-            } else NormalCastTimer -= diff;
+            }
+            else NormalCastTimer -= diff;
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp]: better placing the body of the ``else`` in a new line, for example:

```
else
    NormalCastTimer -= diff;
```

(as well as in other parts of this PR)

File: src/scripts/EasternKingdoms/Karazhan/karazhan.h [hunk 4]
```diff
@@ -1,78 +1,121 @@
...
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/Karazhan/karazhan.h]: this copyright is wrong, we should put scriptdev2, trinitycore, azerothcore as well as xinef if there is still work of him

File: src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp (hunk 2 above)
Comment (talamortis) [inline src/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp]: the block is still used I removed the loop for the axes that he spawns

File: src/scripts/EasternKingdoms/Karazhan/karazhan.h (hunk 4 above)
Comment (talamortis) [inline src/scripts/EasternKingdoms/Karazhan/karazhan.h]: this is all from script development 2 just back ported and made it so it works with azeroth code.

File: src/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp (hunk 3 above)
Comment (talamortis) [inline src/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp]: I will sort this script out later, I was just getting good karazhan to some working progress so it is complete able.

**Comment by DarkmoonRabbit** [comment]
:-) Good job.

**Comment by talamortis** [comment]
@DarkmoonRabbit have you tested fixes?

**Comment by Inifield** [comment]
tested the proposed change and it seems ok to me

**Comment by talamortis** [comment]
@ShinDarth  can we restore this branch so i can finish the work or will it commit to the master again?

**Comment by ShinDarth** [comment]
sure, done

**Comment by talamortis** [comment]
@ShinDarth thanks 👍  still trying to figure out how to use git :P as im just used to fixing stuff and giving to a person to submit lol

**Pull request merged**